src/app/core/adapters/custom-date-adapter.ts

```diff
@@ -0,0 +1,35 @@
+import {NativeDateAdapter} from '@angular/material/core';
+
+export class CustomDateAdapter extends NativeDateAdapter {
+  parse(value: any): Date | null {
+    if (typeof value === 'string' && value.indexOf('/') > -1) {
+      const str = value.split('/');
+
+      const year = Number(str[2]);
+      const month = Number(str[1]) - 1;
+      const date = Number(str[0]);
+
+      return new Date(year, month, date);
+    }
+    const timestamp = typeof value === 'number' ? value : Date.parse(value);
+    return isNaN(timestamp) ? null : new Date(timestamp);
+  }
+
+  format(date: Date, displayFormat: Object): string {
+    date = new Date(
+      Date.UTC(
+        date.getFullYear(),
+        date.getMonth(),
+        date.getDate(),
+        date.getHours(),
+        date.getMinutes(),
+        date.getSeconds(),
+        date.getMilliseconds(),
+      ),
+    );
+    displayFormat = Object.assign({}, displayFormat, {timeZone: 'utc'});
+
+    const dtf = new Intl.DateTimeFormat(this.locale, displayFormat);
+    return dtf.format(date).replace(/[\u200e\u200f]/g, '');
```

Comment:
prefer date-fns methods, we should not start writing custom date handling if avoidable:

https://www.phind.com/search?cache=xd7y1x79v8e56qvqith510z8

same for the format